footer.php

```diff
@@ -332,3 +334,102 @@ function make_navbutton($imgsrc, $bgcolour, $title, $url, $classes = null, $neww
     }
     return $output;
 }
+
+/**
+ * Create Activity completion buttons
+ *
+ * @return string
+ */
+function make_activitycomplete_button() {
+
+    global $COURSE, $DB, $CFG, $PAGE, $OUTPUT;
+
+    $output = "";
+    $style = "";
+
+    if (!$settings = $DB->get_record('navbuttons', array('course' => $COURSE->id))) {
+        return $output;
+    }
+
+    if(!$settings->completebuttonshow) {
+        return $output;
+    }
+
+    $completioninfo = new completion_info($COURSE);
+    $completion = $completioninfo->is_enabled($PAGE->cm);
+    $completiondata = $completioninfo->get_data($PAGE->cm, true);
+
+    $newstate =
+        $completiondata->completionstate == COMPLETION_COMPLETE
+            ? COMPLETION_INCOMPLETE
+            : COMPLETION_COMPLETE;
```

Comment:
Again this is a bit of original functionality taken from core Moodle. However, wrapped condition in () to make it more understandable/readable